flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/DisposableAsync.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+import java.util.concurrent.Executor;
+
+/**
+ * Interface for types that are disposable.
+ *
+ * <p>Some of the disposal work is typically synchronous (like marking the state as disposed) while
+ * some work should be executed asynchronously in the provided executor. Examples for asynchronous disposal
+ * work are heavy or blocking operations involving I/O (like deleting files).
+ */
+public interface DisposableAsync {
+
+	/**
+	 * Disposes this object.
+	 *
+	 * <p>The given executor may be used to schedule expensive/blocking operations where it is
+	 * acceptable that they complete only later.
+	 */
+	void dispose(Executor executor) throws Exception;
```

Comment:
It seems this method is not called anywhere in the patch. Is this intended to be used by the coordinator implementations?

Reply:
This file snuck in. It is part of another follow-up patch (simplifying the trigger procedure). Will remove it while merging.